components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java

```diff
@@ -423,12 +483,23 @@ public Response applicationsApplicationIdKeysKeyTypeGenerateTokenPost(String app
     @Override
     public Response applicationsApplicationIdKeysKeyTypeGet(String applicationId, String keyType,
             String groupId, MessageContext messageContext) {
+        return Response.ok().entity(getApplicationKeyByAppIDAndKeyType(applicationId, keyType)).build();
+    }
 
+    /**
+     * Returns Keys of an application by key type
+     *
+     * @param applicationId Application Id
+     * @param keyType       Key Type (Production | Sandbox)
+     * @return Application Key Information
+     */
+    private ApplicationKeyDTO getApplicationKeyByAppIDAndKeyType(String applicationId, String keyType) {
         List<APIKey> applicationKeys = getApplicationKeys(applicationId);
-        for (APIKey apiKey : applicationKeys) {
-            if (keyType != null && keyType.equals(apiKey.getType())) {
-                ApplicationKeyDTO appKeyDTO = ApplicationKeyMappingUtil.fromApplicationKeyToDTO(apiKey);
-                return Response.ok().entity(appKeyDTO).build();
+        if (applicationKeys!= null) {
```

Comment:
There is a formatting issue

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/6753